src/ImageIO.jl

```diff
@@ -34,12 +34,19 @@ function save(f::File{DataFormat{:PNG}}, image::S; kwargs...) where {T, S<:Union
     end
     return Base.invokelatest(PNGFiles.save, f.filename, image, kwargs...)
 end
-function save(s::Stream{DataFormat{:PNG}}, image::S; kwargs...) where {T, S<:Union{AbstractMatrix, AbstractArray{T,3}}}
+function save(s::Stream{DataFormat{:PNG}}, image::S; permute_horizontal=false, mapi=identity, kwargs...) where {T, S<:Union{AbstractMatrix, AbstractArray{T,3}}}
     if !PNGFiles_LOADED[]
         @eval ImageIO import PNGFiles
         PNGFiles_LOADED[] = isdefined(ImageIO, :PNGFiles)
     end
-    return Base.invokelatest(PNGFiles.save, stream(s), image, kwargs...)
+    imgout = copy(image)
+    if permute_horizontal
+        permutedims!(imgout, imgout, (2,1))
```

Comment:
```julia
julia> x = rand(4, 2)
4×2 Matrix{Float64}:
 0.731366   0.551448
 0.170785   0.106293
 0.0294579  0.460323
 0.200234   0.877563

julia> permutedims!(x, x, (2, 1))
ERROR: DimensionMismatch("destination tensor of incorrect size")
```

To avoid memory allocation, you could instead use `PermutedDimsArray`.

```julia
perms = ndims(image) == 2 ? (2, 1) : ndims(image) == 3 ? (2, 1, 3) : error("$(ndims(image)) dims array is not supported")
imgout = PermutedDimsArray(img, perms)
```

Reply:
Thanks